tests/mir-opt/enum_check_niche.rs

```diff
@@ -0,0 +1,21 @@
+//@ test-mir-pass: CheckEnums
+
+// EMIT_MIR enum_check_niche.main.CheckEnums.diff
+
+#[repr(u16)]
+enum Mix {
+    A,
+    B(u16),
+}
+
+enum Nested {
+    C(Mix),
+    D,
+    E,
+}
+
+fn main() {
+    // CHECK-LABEL: fn main(
+    // CHECK assert(copy .*, "trying to construct an enum from an invalid value {}", .*)
```

Comment:
Why are there mir-opt tests in here? I can't figure out what they are testing for that the UI tests are not.

Also, I think this CHECK is just checking for any enum check after the start of `main`. Writing the FileCheck annotations well seems like a lot of work here. So they should be testing for something that's worth it.

Reply:
If the tests stay in, they need a clear comment in the test that explains what the expected diff is, so that subsequent contributors and reviewers are able to discern if the test "passed" or not when the diff changes.